h2o-algos/src/main/java/hex/gram/Gram.java

```diff
@@ -758,6 +758,34 @@ public void mul(double [] x, double [] res){
     }
   }
 
+  /*
+  This method will not allocate the extra memory and hence is considered for lowMemory systems.
+  However, need to consider case when you have categoricals!  Make them part of the matrix
+  in the multiplication process.
+   */
+  public void mul(double [] x, double [] res, boolean saveMemory){
```

Comment:
The only use of this boolean flag is switching between implementations. I'd suggest to drop that flag, and instead have a function with a different name (mul2 may be not a good one, but well...)